### PR TITLE
Allow unicode or str type for component names

### PR DESCRIPTION
**Pull request by isullivan**

### CODE REVIEW

File: python/lsst/obs/lsstSim/lsstSimMapper.py [hunk 1]
```diff
@@ -134,7 +135,7 @@ def validate(self, dataId):
             if component not in dataId:
                 continue
             val = dataId[component]
-            if not isinstance(val, str):
+            if not isinstance(val, basestring):
```
Comment (PaulPrice) [inline python/lsst/obs/lsstSim/lsstSimMapper.py]: I don't think this is py3-compatible: https://docs.python.org/3.0/whatsnew/3.0.html#text-vs-data-instead-of-unicode-vs-8-bit

Comment (isullivan) [inline python/lsst/obs/lsstSim/lsstSimMapper.py]: I was following https://sqr-014.lsst.io/, and this did pass Jenkins on both python 2 and 3. Is there a different test I should do?

Comment (isullivan) [inline python/lsst/obs/lsstSim/lsstSimMapper.py]: Note the `from past.builtins import basestring` added at the top of the file.

Comment (PaulPrice) [inline python/lsst/obs/lsstSim/lsstSimMapper.py]: I looked for it but didn't see it, but now I see it staring me in the face. Please ignore me. Sorry to have troubled you!

**Pull request merged**